### DISCUSSION TITLE
Pros and Cons of using Eclair vs other implementations (Lnd, C-lightning, etc)?

### DISCUSSION THREAD

**Discussion by dannydeezy** [ACINQ/eclair#2252] [Node operators]
Hi I'm curious to hear people's thoughts on how Eclair compares to the other lightning implementations. My aim is to run a large routing node with lots of channels. I'm super intrigued by Eclair's clustering capabilities, and notice that ACINQ has by far the largest node on the network. Any insights about people's experience with Eclair would be greatly appreciated!

**Comment by pm47** [comment]
_Disclaimer: I'm an eclair developer and operator of the ACINQ node_

If you like to be on the bleeding edge and try out fancy features, or if you need to run a node on constrained hardware, then you should go with respectively lnd and cln. But if your goal is to run a large node with lots of channels, then go with eclair. It has been designed for that purpose from the ground up. Scalability, reliability and performance are top priorities. Eclair is proven to scale to tens of thousands of channels (our node is much larger than what public metrics show, due to the unannounced channels that we have with mobile wallets like Phoenix), and is also the fastest implementation in an [independent benchmark from a year ago](https://github.com/bottlepay/lightning-benchmark).

Crucially, as you noted, the developers of eclair are also the operators of the largest node in the network, that means we [eat our own dog food](https://en.wikipedia.org/wiki/Eating_your_own_dog_food). The ever-growing channels.db issue on lnd is a typical example of something that would never happen on eclair, because it would have killed us. You can also expect flawless updates / db migrations, for the same reason, and a strong focus on reliability in general.

> I'm super intrigued by Eclair's clustering capabilities

The clustering mode is useful to handle many concurrent connected peers and lots of routing table gossip, and also improves security by reducing the exposure of critical parts of the node. We haven't yet worked on clustering the backend part but it will probably happen at some point, eclair's tech stack (akka/scala) makes it pretty easy to do that kind of things.

The main con of eclair is that it runs on the JVM, and that's enough to turn off a lot of people 😆

**Reply by kannapoix** [reply]
@pm47 
>You can also expect flawless updates / db migrations, for the same reason, and a strong focus on reliability in general.

Is it possible with eclair to update eclair without downtime, even with DB schema changes? Can eclair fail over to the other eclair node in the event of a data center disaster?

LND seems supporting fail over with etcd: https://github.com/lightningnetwork/lnd/blob/master/docs/leader_election.md
Are you using a similar setup at ACINQ?

**Comment by rorp** [comment]
@kannapoix  See https://www.youtube.com/watch?v=L2DtolwS8ew

Additionally, taking into account that the typical downtime after a restart for a busy LND node is measured in hour(s), whereas for an Eclair node it is typically just a few seconds, do you really need special arrangements for failover?

**Reply by kannapoix** [reply]
@rorp Thanks, that's helpful!
I see. So you are saying that just a simple fail over configuration by using Load balancer is enough for eclair because it has really quick startup time. I'll test out this and check it works for us.

How about releasing without downtime?
Releasing some kind of changes such as database schema changes are not simple as just release stand by node first and then switch over to it by load balancer, right?

**Reply by rorp** [reply]
I don't think releasing without downtime is possible in current implementation. Something like LDK's phantom node approach could be helpful in this case.

**Reply by kannapoix** [reply]
Hmm, yes, creating a cluster of lightning nodes and treating it as a logical single lightning node can achieve my goal (I think this doesn't work for an LSP node of self-custody wallet because a channel is bind to a single node). As you mentioned, we need a phantom node like approach to receive a payment with any node in the cluster.

I'll put the blog posts like related to the previous video for future readers: https://suredbits.com/category/lightning-201/
